evm-adapters/src/lib.rs

```diff
@@ -242,11 +249,25 @@ mod test_helpers {
             evm.deploy(Address::zero(), compiled.bytecode().unwrap().clone(), 0.into()).unwrap();
 
         let (_, status1, _, _) = evm
-            .call::<(), _, _>(Address::zero(), addr, "greet(string)", "hi".to_owned(), 0.into())
+            .call::<(), _, _>(
+                Address::zero(),
+                addr,
+                "greet(string)",
+                "hi".to_owned(),
+                0.into(),
+                compiled.abi,
```

Comment:
I'd avoid making the abi part of the function signature, wdyt about making it a struct field on the evm?

Reply:
hmmmm so the thing about evm.call is that its explicitly about being able to decode the revert reason: https://github.com/gakonst/foundry/blob/f36c121a7c9b33316aaf51fb44bf4495e3bab933/evm-adapters/src/lib.rs#L98-L99

given that, feels like it is appropriate to pass it in, because the actual evm is agnostic to the error. if you feel strongly i can change tho

Reply:
Imagine that as a user I want to just transact via the evm without knowing anything about the abi, because I use the evm-adapter in a non-forge use case, ideally I don't need to provide any ABI

Reply:
Then I'd argue we should have a function called `call_w_decode` or something

Reply:
I am just kinda against the evm holding any knowledge of any abis